test-requirements.txt

```diff
@@ -4,34 +4,69 @@
 #
 #    pip-compile --upgrade --output-file test-requirements.txt test-requirements.in
 #
-attrs==19.3.0             # via pytest
-bandit==1.6.2             # via -r test-requirements.in
-coverage==5.2.1           # via -r test-requirements.in
-flake8-comprehensions==3.2.3  # via -r test-requirements.in
-flake8-import-order==0.18.1  # via -r test-requirements.in
-flake8-polyfill==1.0.2    # via pep8-naming
-flake8==3.8.3             # via -r test-requirements.in, flake8-comprehensions, flake8-polyfill
-gitdb==4.0.5              # via gitpython
-gitpython==3.1.7          # via bandit
-importlib-metadata==1.7.0  # via flake8, flake8-comprehensions, pluggy, pytest, stevedore
-iniconfig==1.0.1          # via pytest
-mccabe==0.6.1             # via flake8
-more-itertools==8.4.0     # via pytest
-packaging==20.4           # via pytest
-pbr==5.4.5                # via stevedore
-pep8-naming==0.11.1       # via -r test-requirements.in
-pluggy==0.13.1            # via pytest
-py==1.9.0                 # via pytest
-pycodestyle==2.6.0        # via flake8, flake8-import-order
-pyflakes==2.2.0           # via flake8
-pyparsing==2.4.7          # via packaging
-pytest==6.0.1             # via -r test-requirements.in
-pyyaml==5.3.1             # via bandit
-six==1.15.0               # via bandit, packaging
-smmap==3.0.4              # via gitdb
-stevedore==3.2.0          # via bandit
-toml==0.10.1              # via pytest
-zipp==3.1.0               # via importlib-metadata
+attrs==20.3.0
+    # via pytest
+bandit==1.7.0
+    # via -r test-requirements.in
+coverage==5.3.1
+    # via -r test-requirements.in
+flake8-comprehensions==3.3.1
+    # via -r test-requirements.in
+flake8-import-order==0.18.1
+    # via -r test-requirements.in
+flake8-polyfill==1.0.2
+    # via pep8-naming
+flake8==3.8.4
+    # via
+    #   -r test-requirements.in
+    #   flake8-comprehensions
+    #   flake8-polyfill
+gitdb==4.0.5
+    # via gitpython
+gitpython==3.1.12
+    # via bandit
+iniconfig==1.1.1
+    # via pytest
+mccabe==0.6.1
+    # via flake8
+mypy-extensions==0.4.3
+    # via mypy
+mypy==0.800
+    # via -r test-requirements.in
+packaging==20.8
+    # via pytest
+pbr==5.5.1
+    # via stevedore
+pep8-naming==0.11.1
+    # via -r test-requirements.in
+pluggy==0.13.1
+    # via pytest
+py==1.10.0
+    # via pytest
+pycodestyle==2.6.0
+    # via
+    #   flake8
+    #   flake8-import-order
+pyflakes==2.2.0
+    # via flake8
+pyparsing==2.4.7
+    # via packaging
+pytest==6.2.2
+    # via -r test-requirements.in
+pyyaml==5.4.1
+    # via bandit
+six==1.15.0
+    # via bandit
+smmap==3.0.5
+    # via gitdb
+stevedore==3.3.0
+    # via bandit
+toml==0.10.2
+    # via pytest
+typed-ast==1.4.2
```

Comment:
This one breaks the `pypy3` build… I'd love to keep that build, but I'd hate to create a myriad of different requirement files for it all and maybe not run `mypy` for some. Or maybe run the lints and checks in a separate thingy (and only once, instead of for every Python version) :thinking:

Reply:
That seems to work, at least, let's try that for now.